### DISCUSSION TITLE
2.11 BETA - Object journaling support (#151)

### DISCUSSION THREAD

**Discussion by teixemf** [netbox-community/netbox#6096] [Q&A] [answered]
Hi, the Journal option is not available for interfaces, ports and power outlets.
Is this by design?

**Comment by jeremystretch** [comment, marked as answer]
Yes; journaling is currently supported only for [primary models](https://netbox.readthedocs.io/en/feature/development/models/#primary-models). I'm curious, what is your use case for journaling support on individual components?

**Reply by teixemf** [reply]
Thanks for the reply @jeremystretch 
Journaling supported for primary models only can actually be enough. Despite the use case I came up with bellow, at the end of the day it can all be resumed to the level of detail someone want to work with.

On Interfaces, I can see Journaling useful to register the reason which motivated it's change. For instance, the reason about 
why some VLAN was added/deleted/updated on the interface, the reason why was the interface enabled/disabled or, leveraging the journal entries in Markdown Language, a link to the Ticket/Incident/Proposal/Request where the Change was registered.

The Journal of an Interface would show the Interface's history while the Journal of the Device would show not only the Device´s history but also all it's Interfaces history combined together on the same view. Because Interfaces are part of the Device, a change on an Interface is a change on the Device itself. A toggle button could be used to filter out the entries related to the Interfaces from the Device's Journal View.

As for Ports and Power Outlets, I mentioned them mainly for the sake of coherence. However, I can see Journaling being useful for registering maintenance events, specially for Ports.
